### PR TITLE
[Pods] Adding pod actions to pod-specific views

### PR DESCRIPTION
**Pull request by wavesoft**
---
~~⚠️  Depends on #1071~~
~~⚠️  Depends on #1070~~

---

This PR introduces the action buttons and links them to the appropriate pod-specific store actions.

_WIP: Using this PR as a base in order to integrate the `ServiceFormModal` in a separate PR_

### CODE REVIEW

**Comment by philipnrmn** [comment]
I'm happy with the direction of this PR. I haven't done a line-by-line review as it's obviously still WIP and there's nothing more annoying than a close interrogation of unfinished work. Is it in a hold-merge state because of its (recently resolved) PR dependencies or will there be more commits on this branch?

**Comment by mlunoe** [comment]
I tried rebasing this for you, but ran into conflicts. I'll let you take care of this guy

File: src/js/constants/PodActionItem.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const PodActionItem = {
+  EDIT: 'edit',
+  DESTROY: 'destroy',
+  SCALE: 'scale',
+  SUSPED: 'suspend'
```
Comment (philipnrmn) [inline src/js/constants/PodActionItem.js]: s/SUSPED/SUSPEND/

File: src/js/components/PodDetail.js [hunk 2]
```diff
@@ -21,14 +28,41 @@ class PodDetail extends mixin(TabsMixin) {
...
+      activeActionDialog: null
     };
 
     METHODS_TO_BIND.forEach((method) => {
       this[method] = this[method].bind(this);
     });
   }
 
+  handleAction(action) {
+    switch (action.id) {
```
Comment (philipnrmn) [inline src/js/components/PodDetail.js]: Not a big fan of using an object arg with a single parameter here, it feels like more complexity than we need. 

If we made the action type the first name, we could bind in the JSX like
```onClick={props.handleAction.bind(null, PodActionItem.DESTROY)}```

But I'd actually prefer individual explicit handleDestroy/handleEdit/handleScale/handleSuspend  methods.

File: src/js/components/PodDetail.js [hunk 3]
```diff
@@ -56,17 +90,36 @@ class PodDetail extends mixin(TabsMixin) {
...
     const {pod} = this.props;
+    const {activeActionDialog} = this.state;
 
     return (
       <div className="flex-container-col">
         <div className="container-pod container-pod-divider-bottom-align-right container-pod-short-top flush-bottom flush-top media-object-spacing-wrapper media-object-spacing-narrow">
           <Breadcrumbs />
           <PodHeader
-            onDeploy={this.onActionDeploy}
+            onAction={this.handleAction}
```
Comment (philipnrmn) [inline src/js/components/PodDetail.js]: I think I'd prefer to have individual methods than one overloaded method.

File: src/js/components/PodDetail.js (hunk 2 above)
Comment (MatApple) [inline src/js/components/PodDetail.js]: +1

File: src/js/components/PodDetail.js (hunk 3 above)
Comment (wavesoft) [inline src/js/components/PodDetail.js]: Yeah, I also aggree on this. I will have a look on the ticket.

**Pull request merged**